### PR TITLE
Markers Change Shape When Starred

### PR DESCRIPTION
**Pull request by carolinecreamer**
In this commit, markers now change shape when the "star" icon is clicked.

Changes:
- PopoverTrigger component now calls the DB to determine if the city is "starred" or not
- User can see their favorited cities on the map
- Add and remove functions in citiesRoutes.jsx now calls res.send

Next steps:
- User Feed
- Ability to users
- Update user auth

Callouts for Review:
- UI for popover next to Marker looks weird
- Input on user auth

Resources Used:
- [Express Docs](https://expressjs.com/en/5x/api.html#res.sendStatus)
- [API Call Naming Conventions](https://restfulapi.net/resource-naming/)

Before: 
Screen Shot 2022-07-20 at 1 52 41 PM
180080350-c0de41bd-aec9-49c1-837c-0a12f4805eea

After:
Screen Shot 2022-07-21 at 4 47 25 PM
Screen Shot 2022-07-21 at 4 47 16 PM

### CODE REVIEW

File: api/userRoutes/users.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const Parse = require('parse/node')
+
+// Get list of favorited cities
+router.get("/", async (req, res, next) => {
+    try {
+        const query = new Parse.Query(Parse.User);
+
+        const users = await query.find();
+        //const user = await User.getUser();
+        //let cities = user.get("cities");
```
Comment (iwen-fb) [inline api/userRoutes/users.js]: should this be uncommented?

File: ui/src/components/PopoverTrigger/PopoverTrigger.jsx [hunk 2]
```diff
@@ -6,63 +6,93 @@ import Popover from 'react-bootstrap/Popover';
...
+            <g
+                fill="#1266F1"
+                stroke="#fff"
+                strokeWidth="1"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                transform="translate(-12, -24)"
+            >
+                <circle cx="12" cy="10" r="3" />
+                <path d="M12 21.7C17.3 17 20 13 20 10a8 8 0 1 0-16 0c0 3 2.7 6.9 8 11.7z" />
```
Comment (iwen-fb) [inline ui/src/components/PopoverTrigger/PopoverTrigger.jsx]: what do these numbers mean? in general, whenever you have numbers like this floating around - put them in a const with a descriptive name

File: api/userRoutes/users.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+const router  = express.Router();
+const User  = require("../models/models")
+const Parse = require('parse/node')
+
+// Get list of favorited cities
+router.get("/", async (req, res, next) => {
```
Comment (iwen-fb) [inline api/userRoutes/users.js]: If this route is supposed to get favorited cities - i think it should be renamed. in general, no route should be just "/"

**Review by iwen-fb — COMMENTED** [review]
Callouts for Review:

> UI for popover next to Marker looks weird

It looks ok to me according to the pr summary.. don't worry about this too much now - you can look into it once you start working on loading the city data into the popover 

> Input on user auth

File: api/userRoutes/users.js (hunk 1 above)
Comment (carolinecreamer) [reply api/userRoutes/users.js]: Meant to delete! Sorry!

File: ui/src/components/MarkerShape/MarkerShape.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            <g
+                fill="none"
+                stroke="#FF5533"
+                strokeWidth="2"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                transform="translate(-12, -24)"
+            >
+                <circle cx="12" cy="10" r="3" />
+                <path d="M12 21.7C17.3 17 20 13 20 10a8 8 0 1 0-16 0c0 3 2.7 6.9 8 11.7z" />
```
Comment (iwen-fb) [inline ui/src/components/MarkerShape/MarkerShape.jsx:22]: reminder! Please put this into a const variable - other than that, looks good

**Review by iwen-fb — APPROVED** [review]

**Pull request merged**